### PR TITLE
[1LP][RFR] Pre commit black isort

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
this applies `black` and `isort` to `artifactor`, `wt_miq` and a few global python files

the line length for both tools has been adapted to our 100

while doing the change i noted a few py3 compat issues, i put most on ignore

i fixed the one using sorted with the cmp variant and can extract it to a own pr if needed

the others should be followup prs

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 1e37130333697129f09753afc9ea02908b1b5bb1:

`artifactor/plugins/reporter.py`:
- :heavy_exclamation_mark: Line 320:25: **W503** *line break before binary operator*
- :heavy_exclamation_mark: Line 321:25: **W503** *line break before binary operator*

`widgetastic_manageiq/expression_editor.py`:
- :heavy_exclamation_mark: Line 282:13: **W503** *line break before binary operator*
- :heavy_exclamation_mark: Line 283:13: **W503** *line break before binary operator*
- :heavy_exclamation_mark: Line 284:13: **W503** *line break before binary operator*

Hmmm
:scream: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 41e4312aba92794687dca00b82d70438d9d25661:

`artifactor/plugins/reporter.py`:
- :heavy_exclamation_mark: Line 320:25: **W503** *line break before binary operator*
- :heavy_exclamation_mark: Line 321:25: **W503** *line break before binary operator*

`widgetastic_manageiq/expression_editor.py`:
- :heavy_exclamation_mark: Line 314:13: **W503** *line break before binary operator*
- :heavy_exclamation_mark: Line 315:13: **W503** *line break before binary operator*
- :heavy_exclamation_mark: Line 316:13: **W503** *line break before binary operator*

Hmmm
:scream: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 434797517d561a31dc01a8e44aade6038d6dd08d:

`artifactor/plugins/reporter.py`:
- :heavy_exclamation_mark: Line 320:25: **W503** *line break before binary operator*
- :heavy_exclamation_mark: Line 321:25: **W503** *line break before binary operator*

`widgetastic_manageiq/expression_editor.py`:
- :heavy_exclamation_mark: Line 314:13: **W503** *line break before binary operator*
- :heavy_exclamation_mark: Line 315:13: **W503** *line break before binary operator*
- :heavy_exclamation_mark: Line 316:13: **W503** *line break before binary operator*

Hmmm
:scream: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

File: .isort.cfg [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[settings]
+line_length=100
+known_third_party = Crypto,IPython,OpenSSL,PIL,_pytest,attr,bottle,bugzilla,cStringIO,cached_property,click,coverage,cycler,dateparser,dateutil,debtcollector,deepdiff,diaper,docker,dockerbot,dump2polarion,fauxfactory,freeze,gevent,git,github,glanceclient,humanfriendly,importscan,iso8601,jenkins,jinja2,jira,jsmin,keystoneauth1,kwargify,lxml,lya,manageiq_client,matplotlib,miq_version,navmazing,novaclient,numpy,paramiko,pathlib2,pdfminer,pika,pkg_resources,py,pygal,pytesseract,pytest,pytz,requests,riggerlib,rope,scp,selenium,sentaku,setuptools,six,slumber,sphinx,sphinx_rtd_theme,sqlalchemy,tabulate,tzlocal,urllib2,wait_for,werkzeug,widgetastic,widgetastic_patternfly,wrapanapi,xmltodict,yaml,yaycl,yaycl_crypt,zmq
```
Comment (mshriver) [inline .isort.cfg]: Do we have to hand manage this now if we add new requirements?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -2,33 +2,31 @@ exclude: ^(data|sprout)/
...
-# - repo: https://github.com/asottile/seed-isort-config
-#   rev: v1.2.0
-#   hooks:
-#   - id: seed-isort-config
-#     args: [--exclude, data/.*]
+- repo: https://github.com/asottile/seed-isort-config
+  rev: v1.2.0
+  hooks:
+  - id: seed-isort-config
+    args: [--exclude, "^data/.*|^scripts/latest_template_tester_report.py|^sprout/|^scripts/gh-find.py"]
```
Comment (mshriver) [inline .pre-commit-config.yaml]: Why these specific modules/directories? Please include comment in-line.

File: .isort.cfg (hunk 1 above)
Comment (RonnyPfannschmidt) [reply .isort.cfg]: its pre-commit hook managed

File: .isort.cfg [hunk 3]
```diff
@@ -1,3 +1,4 @@
 [settings]
 line_length=100
+# NOTE: never change known_third_party by hand its managed by a hook
```
Comment (mshriver) [inline .isort.cfg]: :+1:

File: artifactor/plugins/reporter.py [hunk 4]
```diff
@@ -267,153 +264,203 @@ def build_dict(self, path, container, contents):
...
                 label = '<span class="label label-{}">{}</span>'.format(
-                    bimdict[v['outcomes']['overall']], v['outcomes']['overall'].upper())
-                proc_name = process_pytest_path(v['name'])[-1]
+                    bimdict[v["outcomes"]["overall"]], v["outcomes"]["overall"].upper()
+                )
+                proc_name = process_pytest_path(v["name"])[-1]
                 link = (
-                    '<a href="#{}">{} {} {} <span style="color:#888888"><em>[{}]</em></span></a>'
-                    .format(v['name'], proc_name, teststring, label, pretty_time))
+                    '<a href="#{}">{} {} {} ' '<span style="color:#888888"><em>[{}]</em></span></a>'
```
Comment (mshriver) [inline artifactor/plugins/reporter.py]: This doesn't look quite right, why is it injecting single quotes here

File: widgetastic_manageiq/expression_editor.py [hunk 5]
```diff
@@ -102,10 +102,9 @@ class date_relative_form_view(View):  # noqa
     ROOT = "//div[@id='exp_editor_div']"
     MAKE_BUTTON = "//span[not(contains(@style,'none'))]//img[@alt='{}']"
     ATOM_ROOT = "./div[@id='exp_atom_editor_div']"
-    EXPRESSIONS_ROOT = VersionPick({
-        Version.lowest(): "./fieldset/div",
-        "5.10": ".//div[@class='panel-body']"
-    })
+    EXPRESSIONS_ROOT = VersionPick(
+        {Version.lowest(): "./fieldset/div", "5.10": ".//div[@class='panel-body']"}
```
Comment (mshriver) [inline widgetastic_manageiq/expression_editor.py]: This is unfortunate, these are very nice to see on individual lines to be able to quickly visually parse which versions are supported by the pick.

File: artifactor/plugins/reporter.py (hunk 4 above)
Comment (RonnyPfannschmidt) [reply artifactor/plugins/reporter.py]: double quotes in html text

**Comment by mshriver** [comment]
LGTM, there is a  lot here though, definitely want other eyes on it.

File: artifactor/plugins/reporter.py (hunk 4 above)
Comment (izapolsk) [reply artifactor/plugins/reporter.py]: please explain

Comment (RonnyPfannschmidt) [reply artifactor/plugins/reporter.py]: actually i missunderstood what was happening at first glance
this is potentially a bug, but could aslo be a rebase artifact

**Pull request merged**